cmd/capibmadm/cmd/powervs/image/import.go

```diff
@@ -0,0 +1,209 @@
+		return importimage(cmd.Context(), imageImportOption)
+	}
+
+	options.AddCommonFlags(cmd)
+	return cmd
+}
+
+func importimage(ctx context.Context, imageImportOption imageImportOptions) error {
+	log := logf.Log
+	log.Info("Importing PowerVS images: ", "service-instance-id", options.GlobalOptions.ServiceInstanceID)
+
+	accountID, err := utils.GetAccountID(ctx)
+	if err != nil {
+		return err
+	}
+
+	sess, err := powervs.NewPISession(accountID, options.GlobalOptions.PowerVSZone, options.GlobalOptions.Debug)
+	if err != nil {
+		return err
+	}
+	// By default Bucket Access is private
+	bucketAccess := "private"
+
+	if imageImportOption.Public {
+		bucketAccess = "public"
+	}
+
+	body := &models.CreateCosImageImportJob{
+		BucketAccess:  &bucketAccess,
+		BucketName:    &imageImportOption.BucketName,
+		ImageFilename: &imageImportOption.ImageFilename,
+		ImageName:     &imageImportOption.ImageName,
+		AccessKey:     imageImportOption.AccessKey,
+		SecretKey:     imageImportOption.SecretKey,
+		Region:        &imageImportOption.Region,
+		StorageType:   strings.ToLower(imageImportOption.StorageType),
+	}
+	imageClient := powerClient.NewIBMPIImageClient(ctx, sess, options.GlobalOptions.ServiceInstanceID)
+	jobRef, err := imageClient.CreateCosImage(body)
+	if err != nil {
+		return err
+	}
+
+	start := time.Now()
+	pollErr := wait.PollImmediate(2*time.Minute, imageImportOption.WatchTimeout, func() (bool, error) {
+		jobclient := powerClient.NewIBMPIJobClient(ctx, sess, options.GlobalOptions.ServiceInstanceID)
+		job, err := jobclient.Get(*jobRef.ID)
+		if err != nil {
+			return false, err
+		}
+
+		if *job.Status.State == "completed" {
+			return true, nil
+		}
+		if *job.Status.State == "failed" {
+			return false, fmt.Errorf("image import job failed to complete, err: %v", job.Status.Message)
+		}
+		log.Info("Image Import Job in-progress,", "current state", job.Status.State)
+		return false, nil
+	})
+	if pollErr == wait.ErrWaitTimeout {
+		pollErr = fmt.Errorf("timed out while waiting for image import job to complete")
+	}
+
+	if pollErr != nil {
+		return fmt.Errorf("image import job failed to complete, err: %v", pollErr)
+	}
+
+	log.Info("Retrieving image details")
+	images, err := imageClient.GetAll()
+	if err != nil {
+		return err
+	}
+	var image *models.ImageReference
+	for _, img := range images.Images {
+		if *img.Name == imageImportOption.ImageName {
+			image = img
+			break
+		}
+	}
+
+	if !imageImportOption.Watch {
+		msg := fmt.Sprintf("Importing Image %s is currently in %s state, Please check the Progress in the IBM Cloud UI\n", *image.Name, *image.State)
+		log.Info(msg)
+		return nil
+	}
+
+	pollErr = wait.PollImmediate(2*time.Minute, imageImportOption.WatchTimeout, func() (bool, error) {
+		img, err := imageClient.Get(*image.ImageID)
+		if err != nil {
+			return false, err
+		}
+		if img.State == "active" {
+			return true, nil
+		}
+		log.Info("Import in-progress, current state: %s", img.State)
+		return false, nil
+	})
```

Comment:
This code needs to be revisited because we are already monitoring the the job created in the above block, so with that do we really need this additional wait

Reply:
Removed this additional wait  as this is not needed after monitoring the job created.